parser/src/aspects/call.rs

```diff
@@ -224,13 +233,28 @@ impl<'a> CallAspect<'a> for Parser<'a> {
     }
 }
 
-impl<'a> Parser<'a> {
-    fn call_with_path(&mut self, path: Vec<&'a str>) -> ParseResult<Expr<'a>> {
-        let callee = self.call_argument()?;
-        let tparams = self.parse_type_parameter_list()?.0;
-        self.call_arguments(path, callee, tparams)
+fn check_emptiness<T>(
+    msg: &str,
+    elements: Vec<T>,
```

Comment:
I see a world where it makes sense to take ownership of the data to ensure that it will not be used later. It would be nice to document that, as the name of the function does not suggest that. Maybe rename it to `ensure_empty` or `fail_if_some`?